lib/matplotlib/legend.py

```diff
@@ -395,6 +395,11 @@ def __init__(self, parent, handles, labels,
         # trim handles and labels if illegal label...
         _lab, _hand = [], []
         for label, handle in zip(labels, handles):
+            if not isinstance(label, str):
+                cbook.warn_deprecated(
+                   "3.3",
+                   message="Passing non-string objects as legend "
```

Comment:
Can you add a note suggesting that they probably wanted to pass in two lists?

Reply:
I'd rather not. This check is now implemented in `Legend` with signature `Legend(parent, handles, labels, ..)`. So the "two lists" does not apply anymore. Otherwise I'd have to make the check in `mlegend._parse_legend_args`. But then the parameter in `Legend` is not checked explicitly. Furthermore, you can pass `labels` positionally or by kwarg; while the str-check would apply to both, the proposed addition would only make sense for the former.

On the other hand, IMHO it's clear enough to respond on a call `legend(lines)` with the message `Passing non-string objects as legend labels is deprecated`. It's rather obvious that we interpret the only parameter as labels.

Reply:
I suspect that we are still going to get some confused users, but this is certainly clearer than it currently is!